concrete/elements/navigation/application_menu.php

```diff
@@ -3,45 +3,83 @@
 
 if (!empty($top)) {
     ?>
-    <div class="<?php if (isset($wrapperClass) && $wrapperClass) { ?><?=$wrapperClass?><?php } ?>">
+    <div class="dashboard-nav-rendered <?php if (isset($wrapperClass) && $wrapperClass) {
+        ?><?=$wrapperClass; ?><?php
+    } ?>">
         <?php if (isset($title) && $title) {
-            ?><h3><?=$title?></h3><?php
-        } ?>
+        ?><h3><?=$title; ?></h3><?php
+    } ?>
         <ul class="nav flex-column">
             <?php
-            $walk = function ($pages) use (&$walk, &$view) {
-                $n = count($pages);
-                if ($n > 0) {
-                    for ($i = 0; $i < $n; $i++) {
-                        $page = $pages[$i];
-                        $next = ($i + 1 < $n) ? $pages[$i + 1] : null;
-                        ?>
-                        <li class="<?=$view->controller->getMenuItemClass($page)?>">
-                            <a href="<?=$page->getCollectionLink()?>"><?=t($page->getCollectionName())?></a>
-                            <?php
-                            if ($view->controller->displayChildPages($page)) {
-                                $children = $view->controller->getChildPages($page);
-                                if (!empty($children)) {
-                                    ?>
-                                    <ul>
-                                    <?php $walk($children); ?>
-                                    </ul><?php
-                                }
-                            }
-                            ?>
+            $n = count($top);
+    if ($n > 0) {
+        for ($i = 0; $i < $n; ++$i) {
+            $page = $top[$i];
+            $next = ($i + 1 < $n) ? $top[$i + 1] : null;
+            $renderTopLevelMenu = $view->controller->displayTopLevelMenu($page);
+            $welcomeID = 185;
+            $c = Page::getCurrentPage(); ?>
+                <li class="<?=$view->controller->getMenuItemClass($page); ?>">
+                  <?php if ($renderTopLevelMenu == 1) {
+                ?>
+                    <a href="<?=$page->getCollectionLink(); ?>"><?=t($page->getCollectionName()); ?></a>
+                  <?php
+            } else {
+                ?>
+                    <a href="#" data-launch-sub-panel-url="<?=URL::to('/ccm/system/panels/dashboard/load_menu'); ?>"
+                    data-load-menu="<?=$page->getCollectionId(); ?>">
+                    <?=$page->getCollectionName(); ?>
+                    </a>
+                  <?php
+            } ?>
+                  <?php
+                    $myAccount = 192;
+            $editAccount = 193;
+            $waitingForMe = 186;
```

Comment:
This isn't going to work. We can't hard coded IDs for dashboard pages here. We have to use `getCollectionPath()` but even then I'm not sure that's the right approach given what i'm seeing here.

Reply:
I'll switch them out for getCollectionPath.  There probably is a better approach to this all together, I'll keep working on it.

Reply:
Hard coded ID's swapped for getCollectionPath: https://github.com/concrete5/concrete5/pull/8644/commits/d48a815e6be29f462737af505192aa2dbe46dcda